phd-tests/tests/src/cpuid.rs

```diff
@@ -63,77 +63,34 @@ async fn cpuid_boot_test(ctx: &Framework) {
     // host's CPUID leaves, then filter to just a handful of leaves that
     // advertise features to the guest (and that Linux guests will check for
     // during boot).
-    let mut host_cpuid = cpuid_utils::host_query_all();
-    info!(?host_cpuid, "read host CPUID");
-    let leaf_0_values = *host_cpuid
-        .get(CpuidIdent::leaf(0))
-        .expect("host CPUID leaf 0 should always be present");
+    let mut host_cpuid = cpuid_utils::host::query_complete(
+        cpuid_utils::host::CpuidSource::BhyveDefault,
+    )?;
+
+    info!(?host_cpuid, "read bhyve default CPUID");
 
     // Linux guests expect to see at least a couple of leaves in the extended
     // CPUID range. These have vendor-specific meanings. This test only encodes
     // AMD's definitions, so skip the test if leaf 0 reports any other vendor.
-    let Ok(CpuidVendor::Amd) = CpuidVendor::try_from(leaf_0_values) else {
+    if host_cpuid.vendor() != CpuidVendor::Amd {
         phd_skip!("cpuid_boot_test can only run on AMD hosts");
-    };
-
-    // Report up through leaf 7 to get the extended feature flags that are
-    // defined there.
-    if leaf_0_values.eax < 7 {
-        phd_skip!(
-            "cpuid_boot_test requires the host to report at least leaf 7"
-        );
     }
 
-    // Keep only standard leaves up to 7 and the first two extended leaves.
-    // Extended leaves 2 through 4 will be overwritten with a fake brand string
-    // (see below).
-    host_cpuid.retain_by_leaf(|leaf| {
-        leaf <= 0x7 || (0x8000_0000..=0x8000_0001).contains(&leaf)
-    });
+    // This test works by injecting a fake brand string into extended leaves
+    // 0x8000_0002-0x8000_0004 and seeing if the guest observes that string. For
+    // this to work those leaves need to be present in the host's extended CPUID
+    // leaves.
+    let Some(ext_leaf_0) = host_cpuid.get(CpuidIdent::leaf(EXTENDED_BASE_LEAF))
+    else {
+        phd_skip!("cpuid_boot_test requires extended CPUID leaves");
```

Comment:
idle suggestion: worth making this fail if we can't get at the brand string leaves? these go back to like... the 586, so if they're not present something spectacularly strange is going on that's probably worth exploding over

Reply:
Good idea, done in 5f7f101.